File: notify-trigger-logger.pl
#!/usr/bin/perl

# Program: notify-trigger-logger.pl
# Purpose: read debug_log table and send emails for rows with alerted="N"
# Env: perl5
# Author: James Briggs, USA
# Date: 2018 11 05
# Usage: ./notify-trigger-logger.pl

use strict;
use diagnostics;

use DBI;

   my $DEBUG = 0;

   my $user = 'root';
   my $pw   = 'password';
   my $host = '127.0.0.1';
   my $db   = 'test';

   my $email = 'me@apple.com';
   my $mailer = '/usr/sbin/sendmail -t -oi';

   my $dbh = DBI->connect("dbi:mysql:database=$db;hostname=$host", $user, $pw) or die $DBI::errstr;

   my $sql = 'select id, ts, user, query, note from debug_log where alerted="N"';
   my $sth = $dbh->prepare($sql);
   my $rv = $sth->execute();

   my $sql2 = 'update debug_log set alerted = "Y" where id = ?';
   my $sth2 = $dbh->prepare($sql2);

   while (my ($id, $ts, $user, $query, $note) = $sth->fetchrow_array()) {
      print "$id, $ts, $user, $query, $note\n" if $DEBUG;

      open X, "|$mailer" or die $!;

      print X <<EOD;
To: $email
Subject: mysql-trigger-debugger: $id: $note

Time: $ts
User: $user
Query:
$query
EOD

      close X or die $!;

      my $rv2 = $sth2->execute($id);

      sleep 1;
   }

   $dbh->disconnect();

   exit;

File: notify-trigger-logger.py
#!/usr/bin/python

# Program: notify-trigger-logger.py
# Purpose: read debug_log table and send emails for rows with alerted="N"
# Env: python
# Author: James Briggs, USA
# Date: 2018 11 05
# Usage: ./notify-trigger-logger.py

import MySQLdb
import os

mailer = "/usr/sbin/sendmail" # mailer location
to = "me@apple.com"

db = MySQLdb.connect("localhost", "root", "password", "db")
 
cursor = db.cursor() 
sql = "select id, ts, user, query, note from debug_log where alerted='N'"
cursor.execute(sql)
results = cursor.fetchall() 

for row in results:
    id = row[0]
    # print id

    subject = "notify-trigger-logger.py: %s: %s" % (id, row[4])

    message = """\
To: %s
Subject: %s

Time: %s
User: %s
Query:
%s
""" % (to, subject, row[1], row[2], row[3])

    p = os.popen("%s -t -i" % mailer, "w")
    p.write(message)
    status = p.close()

    cursor2 = db.cursor() 
    sql2 = "update debug_log set alerted='Y' where id = %s" % id
    cursor2.execute(sql2)
    db.commit()
 
db.close()

